server/odc-migrate/src/main/java/com/oceanbase/odc/migrate/jdbc/common/V42412AddDefaultRiskDetectRules.java

```diff
@@ -0,0 +1,137 @@
+import org.springframework.transaction.support.TransactionTemplate;
+
+import com.oceanbase.odc.common.util.JdbcTemplateUtils;
+import com.oceanbase.odc.core.migrate.JdbcMigratable;
+import com.oceanbase.odc.core.migrate.Migratable;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @Author: Lebie
+ * @Date: 2024/2/27 15:38
+ * @Description: []
+ */
+@Slf4j
+@Migratable(version = "4.2.4.12", description = "add default risk level detect rules")
+public class V42412AddDefaultRiskDetectRules implements JdbcMigratable {
+    private static final String DEFAULT_HIGH_RISK_DETECT_RULE_VALUE =
+            "{\"booleanOperator\":\"OR\",\"children\":[{\"expression\":\"ENVIRONMENT_NAME\",\"operator\":\"EQUALS\",\"type\":\"CONDITION\",\"value\":\"${com.oceanbase.odc.builtin-resource.collaboration.environment.prod.name}\"},{\"expression\":\"TASK_TYPE\",\"operator\":\"IN\",\"type\":\"CONDITION\",\"value\":[\"APPLY_PROJECT_PERMISSION\",\"APPLY_DATABASE_PERMISSION\"]}],\"type\":\"CONDITION_GROUP\"}";
+    private static final String DEFAULT_HIGH_RISK_DETECT_NAME = "default high risk level detect rule";
+
+    private TransactionTemplate transactionTemplate;
+    private JdbcTemplate jdbcTemplate;
+    private NamedParameterJdbcTemplate namedParameterJdbcTemplate;
+
+    @Override
+    public void migrate(DataSource dataSource) {
+        this.jdbcTemplate = new JdbcTemplate(dataSource);
+        this.namedParameterJdbcTemplate = new NamedParameterJdbcTemplate(jdbcTemplate);
+        this.transactionTemplate = JdbcTemplateUtils.getTransactionTemplate(dataSource);
+        transactionTemplate.execute(status -> {
+            try {
+                List<Long> organizationIds = listTeamOrganizationIds();
+                if (organizationIds.isEmpty()) {
+                    log.info("no team organization found");
+                    return null;
+                }
+
+                List<InnerRiskDetectRule> detectRulesToInsert = getOrgId2HighRiskLevelIdMap(organizationIds);
+                if (detectRulesToInsert.isEmpty()) {
+                    log.warn("no high risk level found for organization {}", organizationIds);
+                    return null;
+                }
+
+                batchInsertRules(detectRulesToInsert);
+            } catch (Exception e) {
+                log.error("add default risk detect rules failed", e);
+                status.setRollbackOnly();
+                throw new RuntimeException("add default risk detect rules failed", e);
+            }
+            return null;
+        });
+    }
+
+    private List<Long> listTeamOrganizationIds() {
+        String queryTeamOrganization = "select id from iam_organization where type = 'TEAM';";
+        return jdbcTemplate.queryForList(queryTeamOrganization, Long.class);
+    }
+
+    private List<InnerRiskDetectRule> getOrgId2HighRiskLevelIdMap(List<Long> organizationIds) {
+        String sql =
```

Comment:
is it possible if here got an empty organizationIds ?

Reply:
No, I checked the organizationId list before calling this method.